go/parsers/register.go

```diff
@@ -20,7 +20,10 @@ func (r *Register) RequestID() uint64 {
 }
 
 func (r *Register) Options() map[string]any {
-	return map[string]any{}
+	options := make(map[string]any)
+	options["match"] = r.gen.Match().String()
```

Comment:
only add match or invoke in the map if they actually exist in the underlying r.gen